vaultsrv/admin/services.py

```diff
@@ -0,0 +1,101 @@
+from django import forms
+from django.contrib import admin
+from django.utils.translation import ugettext_lazy as _
+
+from vaultsrv.models import (
+    Account, Customer, ServiceGroup, Service, ServiceGroupMembership
+)
+
+
+class AccountIsStaffListFilter(admin.SimpleListFilter):
+    title = _('created by')
+    parameter_name = 'created_by'
+
+    def lookups(self, request, model_admin):
+        return [
+            (a.id, a.user) for a in Account.objects.filter(user__is_staff=True)
+        ]
+
+    def queryset(self, request, queryset):
+        if self.value():
+            return queryset.filter(created_by__pk=self.value())
+
+
+class CustomerModelForm(forms.ModelForm):
+
+    class Meta:
+        model = Customer
+        fields = '__all__'
+        widgets = {
+            'created_by': forms.widgets.Select(attrs={'disabled': True})
+        }
+
+
+class CustomerAdmin(admin.ModelAdmin):
+    model = Customer
+    form = CustomerModelForm
+    search_fields = ('name',)
+    list_filter = (AccountIsStaffListFilter,)
+    list_display = ('name', 'created_by')
+    readonly_fields = ['created']
+
+    @staticmethod
+    def _set_current_account_as_created_by(request):
+        data = getattr(request, request.method).copy()
+        data['created_by'] = request.user.account.pk
```

Comment:
this crashes on first run because the user created by createsuperuser has no related account.

Reply:
Possible solution:
```        
try:
    data['created_by'] = request.user.account.pk
except Account.DoesNotExist:
    data['created_by'] = None
```

Reply:
Good catch, go ahead and add :)